torchvision/prototype/datasets/datapipes.py

```diff
@@ -0,0 +1,29 @@
+from typing import Tuple, IO, Iterator, Union, cast
+
+import torch
+from torchdata.datapipes.iter import IterDataPipe
+
+__all__ = ["FloReader"]
+
+
+class FloReader(IterDataPipe[torch.Tensor]):
+    def __init__(self, datapipe: IterDataPipe[Tuple[str, IO]]) -> None:
+        self.datapipe = datapipe
+
+    def _read_data(self, file: IO, *, dtype: torch.dtype, count: int) -> torch.Tensor:
+        num_bytes_per_value = (torch.finfo if dtype.is_floating_point else torch.iinfo)(dtype).bits // 8
+        chunk_size = count * num_bytes_per_value
+        return torch.frombuffer(bytearray(file.read(chunk_size)), dtype=dtype)
+
+    def _read_scalar(self, file: IO, *, dtype: torch.dtype) -> Union[int, float]:
+        return self._read_data(file, dtype=dtype, count=1).item()
+
+    def __iter__(self) -> Iterator[torch.Tensor]:
+        for _, file in self.datapipe:
+            if self._read_scalar(file, dtype=torch.float32) != 202021.25:
+                raise ValueError("Magic number incorrect. Invalid .flo file")
+
+            width = cast(int, self._read_scalar(file, dtype=torch.int32))
```

Comment:
Yes, that would also work. `cast(int, ...)` is a no-op at runtime, but probably so `int(...)` if the argument is already an `int`.

Reply:
We probably don't need to worry too much about the computation cost of calling `int` twice on a ML script :p

Reply:
OK.
Sorry, these things really make me sad. Half of the code is just annotations or extra bits to appease mypy. Things are getting hard to follow for me.

Reply:
You definitively have a point. Maybe I'm to "used" to stuff like that now, but if you find it harder to follow, than simply go with your suggestion. Readability should be value higher than `mypy` compatibility where ever possible. We have discussed this at length in the past and there are situations where `mypy` complaining is just the symptom of spaghetti code. In those situations, we should refactor instead of silencing it. But this is not one of these situations.

Reply:
TBH I would prefer your original code and use `cast` instead of `int` for the reasons you mentioned above.

It also makes it clear that it's `mypy` being a PITA again.

I'm just frustrated to see all these code obfuscations for benefits that have never been clear to me.